File: flut/flutter/material/divider.py
from typing import Optional, override

from flut._flut.engine import _flut_pack_value, call_dart_static
from flut.flutter.widgets.framework import Widget


class Divider(Widget):
    _flut_type = "Divider"

    def __init__(
        self,
        *,
        key=None,
        height: Optional[float] = None,
        thickness: Optional[float] = None,
        indent: Optional[float] = None,
        endIndent: Optional[float] = None,
        color=None,
        radius=None,
    ):
        super().__init__(key=key)
        self.height = height
        self.thickness = thickness
        self.indent = indent
        self.endIndent = endIndent
        self.color = color
        self.radius = radius

    @override
    def _flut_pack(self) -> dict:
        result = self._flut_base_props()
        if self.height is not None:
            result["height"] = _flut_pack_value(self.height)
        if self.thickness is not None:
            result["thickness"] = _flut_pack_value(self.thickness)
        if self.indent is not None:
            result["indent"] = _flut_pack_value(self.indent)
        if self.endIndent is not None:
            result["endIndent"] = _flut_pack_value(self.endIndent)
        if self.color is not None:
            result["color"] = _flut_pack_value(self.color)
        if self.radius is not None:
            result["radius"] = _flut_pack_value(self.radius)
        return result

    @staticmethod
    def createBorderSide(context, *, color=None, width=None):
        return call_dart_static(
            "Divider",
            "createBorderSide",
            context._flut_pack() if context is not None else None,
            color=color._flut_pack() if color is not None else None,
            width=width,
        )

File: flut/.flutter/lib/flutter/material/divider.dart
import 'package:flutter/material.dart';
import 'package:flut/flut/runtime.dart';

class FlutDivider {
  FlutDivider._();

  static Divider? flutDecode(FlutRuntime runtime, Map<String, dynamic> data) {
    return Divider(
      key: runtime.decodeKey(data),
      height: runtime.unpackOptionalField<double>(data, 'height'),
      thickness: runtime.unpackOptionalField<double>(data, 'thickness'),
      indent: runtime.unpackOptionalField<double>(data, 'indent'),
      endIndent: runtime.unpackOptionalField<double>(data, 'endIndent'),
      color: runtime.unpackOptionalField<Color>(data, 'color'),
      radius: runtime.unpackOptionalField<BorderRadiusGeometry>(data, 'radius'),
    );
  }

  static void registerStatics(FlutRuntime runtime) {
    runtime.registerStatic('Divider.createBorderSide', createBorderSide);
  }

  static dynamic createBorderSide(
    FlutRuntime runtime,
    BuildContext? context, {
    Color? color,
    double? width,
  }) {
    return Divider.createBorderSide(context, color: color, width: width);
  }
}
